### ISSUE TITLE
Error in spec initialization is cryptic with recent intellij

### ISSUE THREAD

**Issue by sksamuel** [kotest/kotest#1097]
```
class FooTest : FunSpec() {
  private val ds = "qweqw".apply { error(this) }
 init {
      test("foo") {
      }
  }
}
```

Gives no indication of the actual exception.

**Comment by sksamuel** [comment]
Seems same is happening with an exception in beforeSpec

**Comment by LeoColman** [comment]
I thought we already fixed this previously

**Comment by LeoColman** [comment]
Did we revert that?

**Comment by sksamuel** [comment]
This is now fixed in 4.0.
